yt_dlp/extractor/youtube.py

```diff
@@ -688,31 +696,51 @@ def _extract_identity_token(self, ytcfg=None, webpage=None):
                 r'\bID_TOKEN["\']\s*:\s*["\'](.+?)["\']', webpage,
                 'identity token', default=None, fatal=False)
 
-    @staticmethod
-    def _extract_account_syncid(*args):
+    def _data_sync_id_to_delegated_session_id(self, data_sync_id):
+        if not data_sync_id:
+            return
+        # datasyncid is of the form "channel_syncid||user_syncid" for secondary channel
+        # and just "user_syncid||" for primary channel. We only want the channel_syncid
+        session_ids = data_sync_id.split('||')
+        if len(session_ids) >= 2 and session_ids[1]:
+            return session_ids[0]
```

Comment:
```suggestion
        channel_syncid, _, user_syncid = data_sync_id.partition('||')
        if user_syncid:
            return channel_syncid
```

Reply:
Is `data_sync_id` always a string? if so, the `if not data_sync_id` could be omitted as well.